File: 3 - algorithms/cards.c
#include <stdio.h>
#include <cs50.h>
void shuffle();
typedef struct
{
    int suit;
    int rank;
} card;

card deck[52];

string ranks[] = {"", "A", "2", "3", "4", "5", "6",
                  "7", "8", "9", "10", "J", "Q", "K"};
string suits[] = {"♥", "♣", "♠", "♦"};

int main(void)
{
    // populate deck
    int n = 0;
    for (int s = 0; s < 4; s++)
    {
        for (int r = 1; r <= 13; r++)
        {
            deck[n].suit = s;
            deck[n].rank = r;
            n++;
        }
    }
    shuffle();
    printf("%s%s\n", ranks[deck[0].rank], suits[deck[0].suit]);
}

void shuffle()
{

}

File: python/cards.py
import random
ranks = ["", "A", "2", "3", "4", "5", "6",
         "7", "8", "9", "10", "J", "Q", "K"]
suits = ["♥", "♣", "♠", "♦"]

class Card:
    def __init__(self, suit, rank):
        self.suit = suit
        self.rank = rank
    
    def __str__(self):
        return f"{ranks[self.rank]}{suits[self.suit]}"

class Deck:
    def __init__(self):
        self.cards = []
        for s in range(4):
            for r in range(1, 14):
                self.cards.append(Card(s, r))
    
    def shuffle(self):
        random.shuffle(self.cards)
        
    def deal_card(self):
        return self.cards.pop()

random.seed(1234)
d = Deck()
d.shuffle()
for i in range(5):
    print(d.deal_card())